### PR TITLE
ADEN-12266 Updating GAM API and adding "Size merge" tasks

### PR DESCRIPTION
**Pull request by dmnsobczak**

### CODE REVIEW

**Review by nandy-andy — APPROVED** [review]

File: src/Report/Api/StatementBuilder.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+{
+	public static function build(ParameterBag $parameters) {
+		$statementBuilder = new AdManagerStatementBuilder();
+		$statements = [];
+		$i = 1;
+		foreach ($parameters->get('filters') as $type => $value) {
+			if ($value === '') {
+				continue;
+			}
+			$filter = ReportService::DIMENSION_MAPPING[$type];
```
Comment (nandy-andy) [inline src/Report/Api/StatementBuilder.php]: Can't we just put the map within the `StatementBuilder` and get rid off the `ReportService`? 🤔

File: src/Report/Api/StatementBuilder.php [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+	/**
+	 * @param $statementBuilder
+	 * @param $index
+	 * @param $filter
+	 * @param $type
+	 * @param $values
+	 * @return string
+	 *
+	 * @TODO currently there is no support for multiple values in not statement
```
Comment (nandy-andy) [inline src/Report/Api/StatementBuilder.php]: This comment except TODO part is useless. Let's remove most of it ;)

Comment (dmnsobczak) [reply src/Report/Api/StatementBuilder.php]: Done!

File: src/Report/Api/StatementBuilder.php (hunk 1 above)
Comment (dmnsobczak) [reply src/Report/Api/StatementBuilder.php]: Done!

File: composer.json [hunk 3]
```diff
@@ -2,15 +2,15 @@
     "license": "Apache-2.0",
     "require": {
         "dflydev/doctrine-orm-service-provider": "^2.0",
-        "googleads/googleads-php-lib": "^57.0",
+        "googleads/googleads-php-lib": "^58.0",
```
Comment (nandy-andy) [inline composer.json:5]: Woo hoo! Thanks for updating the lib! 🙇

**Review by nandy-andy — APPROVED** [review]

**Pull request merged**